lib/types.ml

```diff
@@ -78,22 +78,3 @@ module Opam = struct
 
   let sort_uniq l = List.sort_uniq (fun a b -> String.compare a.name b.name) l
 end
```

Comment:
Wouldn't it make sense to move part of the `Opam` module to `Duniverse` and remove this file?

What is in `Types`:
- `Types.Opam.entry`, `package` and `pin` are used in `Duniverse` in type declarations and be defined there.
- `Types.Opam.t` is not used anywhere and should be removed.
- You added a second module named `Opam` and that may be confusing
- `Types.Opam,Remote` is only used to represent an argument of `Exec.init_opam_and_remotes` and should be defined as a tuple there.
- `load` and `save` functions are not used anymore.
- `sort_uniq` function does not really make sense and should be moved to where it is used.

Reply:
As I mentionned in the PR description, this is just a step in a refactoring, let's try to keep things small here.

Besides that, I think some of those types are used elsewhere and I haven't spent the time to think of a better representation for those so I'd rather do that step by step if you don't mind.

Reply:
To clarify I do believe it would make sense to refine those and move them elsewhere indeed but probably as part of later refactoring!

Reply:
I understand. I still think this is in the scope of this PR as you started removing `Types.Dune` and making some functions unused and you change the way the other are used. Also `Types.Opam` is used in your new representation.
Can you at least remove the things you are making unused by this PR ? (`Types.Opam.t`, `load`, `save`)

Reply:
Hmm yeah right, I'll remove the unused stuff, my bad!